modules/core/02-client/types/msgs.go

```diff
@@ -265,6 +269,32 @@ func (msg *MsgRecoverClient) ValidateBasic() error {
 	return nil
 }
 
+// NewMsgProvideCounterparty creates a new MsgProvideCounterparty instance
+func NewMsgProvideCounterparty(signer, clientID, counterpartyID string) *MsgProvideCounterparty {
+	return &MsgProvideCounterparty{
+		Signer:         signer,
+		ClientId:       clientID,
+		CounterpartyId: counterpartyID,
+	}
+}
+
+// ValidateBasic performs basic checks on a MsgProvideCounterparty.
+func (msg *MsgProvideCounterparty) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.Signer); err != nil {
+		return errorsmod.Wrapf(ibcerrors.ErrInvalidAddress, "string could not be parsed as address: %v", err)
+	}
+
+	if err := host.ClientIdentifierValidator(msg.ClientId); err != nil {
+		return err
+	}
+
+	if strings.TrimSpace(msg.CounterpartyId) == "" {
+		return errorsmod.Wrapf(ErrInvalidCounterparty, "counterparty client id cannot be empty")
```

Comment:
Hmm.... it should be a valid identifier. I don't think it should be client identifier necessarily. because once we add aliasing it might be an alias. But definitely the client validator function works for now.

https://github.com/cosmos/ibc-go/blob/main/modules/core/24-host/validate.go#L65

In the future we can create a new function name that is just Identifier validation or something more generic.